### PR TITLE
Replace rq with celery

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: mergify_engine/worker_old.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# -*- encoding: utf-8 -*-
```
Comment (jd) [inline mergify_engine/worker_old.py]: Was that kept on purpose?

File: mergify_engine/worker_old.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        for q in self.queues:
+            if q.name == job_origin:
+                job = rq.Job()
+                job.data = job_data
+                job.meta['failure'] = failure
+                job.description = job.get_call_string()
+                q.enqueue_job(job, at_front=True)
+
+
+# Bootstrap celery worker
```
Comment (jd) [inline mergify_engine/worker_old.py]: ?

File: mergify_engine/worker_old.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                scheduler = rq_scheduler.Scheduler(queue=q)
+                scheduler.enqueue_in(retry_in, self._retry_now,
+                                     job.origin,
+                                     job.data,
+                                     job.meta['failures'])
+                return False
+
+        # Can't find queue, which should basically never happen as we only
+        work # jobs that match the given queue names and queues are transient
+        in rq.
```
Comment (jd) [inline mergify_engine/worker_old.py]: that sounds invalid :)

File: mergify_engine/tests/test_engine.py [hunk 4]
```diff
@@ -185,6 +185,12 @@ def cleanup(self):
             self.save_records()
 
 
+# NOTE(sileht): Celery magic, this just skip amqp and execute tasks directly
+# So all REST API calls will block and exectute celery tasks directly
```
Comment (jd) [inline mergify_engine/tests/test_engine.py]: execute

**Comment by sileht** [comment]
Yes for the cache

File: mergify_engine/worker_old.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/worker_old.py]: I just have missed to delete this file.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now bda4df21d740c5ab2573ba383f480ba55db06f56

**Comment by sileht** [comment]
Pull-request updated, HEAD is now bcbd424fe48361bb45368c6f6267f8e189ed2db9

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 37092c0eef3d781f5c2873f2c52ec1dabf41fbc8

**Comment by sileht** [comment]
Pull-request updated, HEAD is now bb9bf0edd799bbb37e709f313fcdba572b762303

**Pull request merged**